### ISSUE TITLE
Entry doesn't belong in directory - FTP

### ISSUE THREAD

**Issue by seonghobae** [ncw/rclone#3283]
#### What is the problem you are having with rclone?
```rclone ncdu cran```
cran storage address was ftp://cran.r-project.org/pub/R/

I can access pub/R/bin/macosx/leopard/base/R-2.15.3.pkg (that may symlinked) via web browser however ncdu was not print them.

![image](https://user-images.githubusercontent.com/8172694/59962922-df59a200-9526-11e9-89e4-389715b1a2e0.png)
![image](https://user-images.githubusercontent.com/8172694/59962928-fdbf9d80-9526-11e9-8bdc-07e4d407de20.png)

#### What is your rclone version (output from `rclone version`)
```rclone --version
rclone v1.47.0-019-g3d475dc0-beta
- os/arch: linux/amd64
- go version: go1.12.4
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
```uname -a
Linux seongho-HP 4.15.0-50-lowlatency #54-Ubuntu SMP PREEMPT Mon May 6 19:38:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

####  Which cloud storage system are you using? (eg Google Drive)
FTP

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```rclone ncdu cran```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
$ rclone lsd cran:pub/R/bin/macosx -L -vv
2019/06/22 21:10:25 DEBUG : rclone: Version "v1.48.0-006-g5935cb0a-beta" starting with parameters ["rclone" "lsd" "cran:pub/R/bin/macosx" "-L" "-vv"]
2019/06/22 21:10:25 DEBUG : Using config file from "/home/seongho/.config/rclone/rclone.conf"
2019/06/22 21:10:25 DEBUG : ftp://cran.r-project.org:21/pub/R/bin/macosx: Connecting to FTP server
2019/06/22 21:10:29 ERROR : Mac-GUI-1.64.tar.gz -> old/Mac-GUI-1.64.tar.gz: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : Mac-GUI-1.65.tar.gz -> old/Mac-GUI-1.65.tar.gz: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.3.3.pkg -> old/R-3.3.3.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.4.0.pkg -> el-capitan/base/R-3.4.0.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.4.1.pkg -> el-capitan/base/R-3.4.1.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.4.2.pkg -> el-capitan/base/R-3.4.2.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.4.3.pkg -> el-capitan/base/R-3.4.3.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.4.4.pkg -> el-capitan/base/R-3.4.4.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.5.0.pkg -> el-capitan/base/R-3.5.0.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.5.1.pkg -> el-capitan/base/R-3.5.1.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.5.2.pkg -> el-capitan/base/R-3.5.2.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.5.3.pkg -> el-capitan/base/R-3.5.3.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-3.6.0.pkg -> el-capitan/base/R-3.6.0.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
2019/06/22 21:10:29 ERROR : R-latest.pkg -> el-capitan/base/R-latest.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring
          -1 2016-03-28 09:00:00        -1 contrib
          -1 2017-07-07 09:00:00        -1 el-capitan
          -1 2009-10-28 09:00:00        -1 leopard
          -1 2014-03-27 09:00:00        -1 mavericks
          -1 2018-04-29 09:00:00        -1 old
          -1 2019-04-27 07:10:00        -1 tools
2019/06/22 21:10:29 DEBUG : 3 go routines active
2019/06/22 21:10:29 DEBUG : rclone: Version "v1.48.0-006-g5935cb0a-beta" finishing with parameters ["rclone" "lsd" "cran:pub/R/bin/macosx" "-L" "-vv"]
```

**Comment by ncw** [comment]
> 2019/06/22 21:10:29 ERROR : R-3.4.0.pkg -> el-capitan/base/R-3.4.0.pkg: Entry doesn't belong in directory "" (contains subdir) - ignoring

The problem appears to be that the FTP server is returning `R-3.4.0.pkg -> el-capitan/base/R-3.4.0.pkg` as the name of the file rather than the expected `R-3.4.0.pkg`.  I suspect that is because it is a symlink.

A workaround would be to look in the directory here instead

```
 rclone lsf --ftp-user anonymous --ftp-pass Zahxrn5gXg8MfAWBAwkUDaMoeWg7lHhkRkuZQzPUcR_bawY --ftp-host cran.r-project.org :ftp:/pub/R/bin/macosx/el-capitan/base/
R-3.4.0.pkg
R-3.4.1.pkg
R-3.4.2.pkg
R-3.4.3.pkg
R-3.4.4.pkg
R-3.5.0.pkg
R-3.5.1.pkg
R-3.5.2.pkg
R-3.5.3.pkg
R-3.6.0.pkg
R-latest.pkg -> R-3.6.0.pkg
```

It looks like the FTP server doesn't support the more modern MLSD so rclone falls back to LIST which is not very well standardized.

```
-rw-rw-r--    1 0        1001         1089 Apr 26 22:05 MD5
lrwxrwxrwx    1 0        1001           23 Jul 11  2014 Mac-GUI-1.64.tar.gz -> old/Mac-GUI-1.64.tar.gz
lrwxrwxrwx    1 0        1001           23 Jun 19  2015 Mac-GUI-1.65.tar.gz -> old/Mac-GUI-1.65.tar.gz
-rw-rw-r--    1 0        1001      1517186 Jun 19  2015 Mac-GUI-1.66.tar.gz
-rw-rw-r--    1 0        1001      1760577 Mar 11  2016 Mac-GUI-1.67.tar.gz
-rw-rw-r--    1 0        1001      1795366 May 04  2016 Mac-GUI-1.68.tar.gz
-rw-rw-r--    1 0        1001      1795570 Apr 19  2017 Mac-GUI-1.69.tar.gz
-rw-rw-r--    1 0        1001      1795622 Jul 07  2017 Mac-GUI-1.70.tar.gz
-rw-rw-r--    1 0        1001        73680 Dec 22 22:51 NEWS
-rw-rw-r--    1 0        1001     71805854 Jun 19  2015 R-3.2.1-snowleopard.pkg
lrwxrwxrwx    1 0        1001           15 May 01  2017 R-3.3.3.pkg -> old/R-3.3.3.pkg
lrwxrwxrwx    1 0        1001           27 Jul 07  2017 R-3.4.0.pkg -> el-capitan/base/R-3.4.0.pkg
lrwxrwxrwx    1 0        1001           27 Jul 07  2017 R-3.4.1.pkg -> el-capitan/base/R-3.4.1.pkg
lrwxrwxrwx    1 0        1001           27 Oct 04  2017 R-3.4.2.pkg -> el-capitan/base/R-3.4.2.pkg
lrwxrwxrwx    1 0        1001           27 Dec 07  2017 R-3.4.3.pkg -> el-capitan/base/R-3.4.3.pkg
lrwxrwxrwx    1 0        1001           27 Mar 15  2018 R-3.4.4.pkg -> el-capitan/base/R-3.4.4.pkg
lrwxrwxrwx    1 0        1001           27 Apr 24  2018 R-3.5.0
```

You can see the symlinks in the listing - rclone isn't interpreting those properly.

This needs fixing in https://github.com/jlaffaye/ftp

Or maybe the FTP server admin could enable MLSD which was designed for machine use.

**Comment by ncw** [comment]
I made an upstream issue: https://github.com/jlaffaye/ftp/issues/151 about this

**Comment by seonghobae** [comment]
The jlaffaye/ftp#151 makes fixes and closed issue.

**Comment by ncw** [comment]
I've merged this to master now (thanks @gary-kim ) which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.49
